keyboards/comet46/keymaps/default/keymap.c

```diff
@@ -0,0 +1,159 @@
+  //,----+----+----+----+----+----+               +----+----+----+----+----+----.
+         , 1  , 2  , 3  , 4  , 5  ,                 6  , 7  , 8  , 9  , 0  ,DEL ,
+  //|----+----+----+----+----+----+----+     +----+----+----+----+----+----+----|
+         , F1 , F2 , F3 , F4 , F5 , F6 ,          ,LEFT,DOWN, UP ,RGHT,END ,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+         , F7 , F8 , F9 , F10, F11, F12,      HOME,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+                             ,    ,    ,          ,    ,    
+  //                    +----+----+---/       \---+----+----+
+  ),
+
+  [_LOWER] = LAYOUT_kc(  
+  //,----+----+----+----+----+----+               +----+----+----+----+----+----.
+         ,EXLM, AT ,HASH,DLR ,PERC,                CIRC,AMPR,ASTR,LPRN,RPRN,DEL ,
+  //|----+----+----+----+----+----+----+     +----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,    ,      GRV ,BSLS,MINS,EQL ,LBRC,RBRC,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,    ,      TILD,PIPE,UNDS,PLUS,LCBR,RCBR,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+                             ,    ,    ,          ,    ,    
+  //                    +----+----+---/       \---+----+----+
+  ),
+
+  [_ADJUST] = LAYOUT_kc(  
+  //,----+----+----+----+----+----+               +----+----+----+----+----+----.
+         ,    ,    ,    ,    ,    ,                    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----+     +----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,CAD ,          ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+         ,    ,    ,    ,    ,    ,RST ,          ,    ,    ,    ,    ,    ,    ,
+  //|----+----+----+----+----+----+----|     |----+----+----+----+----+----+----|
+                             ,    ,    ,          ,    ,    
+  //                    +----+----+---/       \---+----+----+
+  )
+};
+
+
+void matrix_scan_user(void) {
+    uint8_t layer = biton32(layer_state);
+    switch (layer) {
+    	case _QWERTY:
+    	    set_led_green;
+    	    break;
+        case _RAISE:
+            set_led_blue;
+            break;
+        case _LOWER:
+            set_led_red;
+            break;
+        case _ADJUST:
+            set_led_magenta;
+            break;
+        default:
+            set_led_green;
+            break;
+    }
+};
+
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```

Comment:
You can use `set_single_persistent_default_layer(_QWERTY);` here instead, as it handles this, and some additional stuff.